main.tf

```diff
@@ -35,30 +40,38 @@ module "nlb" {
       backend_protocol   = "TCP"
       backend_port       = var.port_gateway
       preserve_client_ip = false
+      health_check       = local.health_check
     },
     {
       name               = "${module.this.id}-${var.port_metadata}"
       backend_protocol   = "TCP"
       backend_port       = var.port_metadata
       preserve_client_ip = false
+      health_check       = local.health_check
+
     },
     {
       name               = "${module.this.id}-${var.port_health}"
       backend_protocol   = "TCP"
       backend_port       = var.port_health
       preserve_client_ip = false
+      health_check       = local.health_check
+
     },
     {
-      name               = "${module.this.id}-80"
+      name               = "${module.this.id}-${var.port_grpc}"
       backend_protocol   = "TCP"
-      backend_port       = 80
+      backend_port       = var.port_grpc
       preserve_client_ip = false
+      health_check       = local.health_check
     },
     {
-      name               = "${module.this.id}-${var.port_grpc}"
+      name               = "${module.this.id}-80"
       backend_protocol   = "TCP"
-      backend_port       = var.port_grpc
+      backend_port       = 80
       preserve_client_ip = false
+      health_check       = local.health_check
+
     },
```

Comment:
let's just drop port 80 completely, shouldn't we? In the end nobody should be using unencrypted traffic anyways and it should be redirected to https instead on LB side

Reply:
thats just the target group